src/lib/V2DutchOrderLib.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {OrderInfo} from "../base/ReactorStructs.sol";
+import {DutchOutput, DutchInput, DutchOrderLib} from "./DutchOrderLib.sol";
+import {OrderInfoLib} from "./OrderInfoLib.sol";
+
+struct V2DutchOrderInner {
+    // generic order information
+    OrderInfo info;
+    // The address which must cosign the ful order
+    address cosigner;
+    // The tokens that the swapper will provide when settling the order
+    DutchInput input;
+    // The tokens that must be received to satisfy the order
+    DutchOutput[] outputs;
+}
+
+struct V2DutchOrder {
+    // Inner order
+    V2DutchOrderInner inner;
+    // The time at which the DutchOutputs start decaying
+    uint256 decayStartTime;
+    // The time at which price becomes static
+    uint256 decayEndTime;
+    // The address who has exclusive rights to the order until decayStartTime
+    address exclusiveFiller;
+    // The amount in bps that a non-exclusive filler needs to improve the outputs by to be able to fill the order
+    uint256 exclusivityOverrideBps;
+    // The tokens that the swapper will provide when settling the order
+    uint256 inputOverride;
+    // The tokens that must be received to satisfy the order
+    uint256[] outputOverrides;
```

Comment:
inputOverride is for exactOutput trades, outputOverrides are for exactInput

> if a filler is overriding, theyre going to do so at exactly the bps that are required to override right (because why would they override by more).

note its not up to the filler to override in this case, these are strictly enforced on top of the inner order based on the secondary auction. True that fillers wouldn't do more than required, but that's ok in this case